src/scheduler/server_info.cpp

```diff
@@ -748,11 +750,13 @@ query_server_dyn_res(server_info *sinfo)
 					log_eventf(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"Program %s timed out", conf.dynamic_res[i].command_line);
 					kill(-pid, SIGTERM);
-					if (waitpid(pid, NULL, WNOHANG) == 0) {
-						usleep(250000);
-						if (waitpid(pid, NULL, WNOHANG) == 0) {
-							kill(-pid, SIGKILL);
-							waitpid(pid, NULL, 0);
+					if (pid != 0) {
```

Comment:
I'm not sure this check is currently necessary, since pid will always be set if we're running this code.

Reply:
For this one, yes. But I found it more consistent to have this check in both places.

Reply:
I am always enclined to _directly_ check whether arguments to a system call are sane rather than rely on the code before (in this case that pipe_err was set, so that we don't end up here). 

Frequently, what works without being defensive becomes a bug when you refactor the other code.

But I'll remove it and replace it with a comment (just as a warning that there _is_ an assumption there you'd better not break).